homeassistant/components/nice_go/light.py

```diff
@@ -24,11 +30,19 @@ async def async_setup_entry(
 
     coordinator = config_entry.runtime_data
 
-    async_add_entities(
-        NiceGOLightEntity(coordinator, device_id, device_data.name)
-        for device_id, device_data in coordinator.data.items()
-        if device_data.light_status is not None
-    )
+    entities = []
+
+    for device_id, device_data in coordinator.data.items():
+        if device_data.type in SUPPORTED_DEVICE_TYPES:
+            entities.append(NiceGOLightEntity(coordinator, device_id, device_data.name))
+        elif device_data.type not in KNOWN_UNSUPPORTED_DEVICE_TYPES:
+            _LOGGER.warning(
+                "Device '%s' has unknown device type '%s', which is not supported by the light platform. This device type is unknown, so please create an issue even if your device does not have a light",
+                device_data.name,
+                device_data.type,
+            )
```

Comment:
Please make sure lines are not longer than 88 characters. I think this message can be a bit to the point. I would maybe suggest keeping this list in a more central place rather than the light setup, since in the future you might have more devices that have their own speciality like idk, a physical switch or a binary sensor for detection.

I would add the link to the place where they should report this.

"We found device "x" which is not supported by the integration. Please create an issue at x"

Reply:
I have the same setup in `switch.py` as well. Should I move the supported devices list into const like so?

```python
SUPPORTED_DEVICE_TYPES = {
    "light": ["WallStation"],
    "switch": ["WallStation"],
}
KNOWN_UNSUPPORTED_DEVICE_TYPES = {
    "light": ["WallStation"],
    "switch": ["WallStation"],
}
```

Reply:
I think it would make sense to centralize this list yes. And I think it depends on how much devices you expect to support. If you expect 100 devices, I would go for a dict with platform as key.

```py
DEVICE_PLATFORMS = {
    Platform.COVER: [
        WALL_STATION,
        MMS100,
    ],
    Platform.SWITCH: [
        WALL_STATION,
    ]
```
and then you can also create a `set` with all the device types that you know, since all devices that you support at least 1 platform. 
```py
SUPPORTED_DEVICES = {dev for dev in devices for _, devices in DEVICE_PLATFORMS}
```

If you only expect to support a few devices or device lines, I would suggest doing it by model -> platform

Reply:
There's only about 6 device types according to the app:

![image](https://github.com/user-attachments/assets/325b498e-49e9-4284-8864-daa4beea4367)
![image](https://github.com/user-attachments/assets/3c3164b7-812e-4828-b2d4-35797b0bc046)

Should I implement what I suggested then? Or am I misinterpreting what you're saying?

Reply:
That's fine, I don't have a strong preference, but I just wanted to say that I would have it depend on the amount of devices